### PR TITLE
[SYCL] Remove redundant event dependencies for in-order queues

### PR DESCRIPTION
**Pull request by sergey-semenov**
Avoid passing explicit dependencies on events from the same in-order
queue since the order of execution is already guaranteed in that case.

### CODE REVIEW

File: sycl/source/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -1270,6 +1283,10 @@ ContextImplPtr MemCpyCommandHost::getContext() const {
   return detail::getSyclObjImpl(Queue->get_context());
 }
 
+QueueImplPtr MemCpyCommandHost::getWorkerQueue() const {
+  return MQueue->is_host() ? MSrcQueue : MQueue;
+}
+
 cl_int MemCpyCommandHost::enqueueImp() {
   QueueImplPtr Queue = MQueue->is_host() ? MSrcQueue : MQueue;
```
Comment (s-kanaev) [inline sycl/source/detail/scheduler/commands.cpp:1291]: NIT: Should we use `getWorkerQueue()` where possible now?

File: sycl/unittests/scheduler/InOrderQueueDeps.cpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  // Check that sequential memory movements submitted to the same in-order
+  // queue do not depend on each other.
+  detail::Command *Cmd = MS.insertMemoryMove(Record, &Req, DefaultHostQueue);
+  detail::EnqueueResultT Res;
+  MockScheduler::enqueueCommand(Cmd, Res, detail::NON_BLOCKING);
+  Cmd = MS.insertMemoryMove(Record, &Req, InOrderQueueImpl);
+  MockScheduler::enqueueCommand(Cmd, Res, detail::NON_BLOCKING);
+  Cmd = MS.insertMemoryMove(Record, &Req, DefaultHostQueue);
+  MockScheduler::enqueueCommand(Cmd, Res, detail::NON_BLOCKING);
```
Comment (s-kanaev) [inline sycl/unittests/scheduler/InOrderQueueDeps.cpp:128]: The test only employs memory-copy commands, isn't it? Should the test be augmented with arbitrary different type of command?

**Review by s-kanaev — COMMENTED** [review]
Question: is there any corner case when alloca command isn't present in dependencies of dependent command?

File: sycl/source/detail/scheduler/commands.cpp [hunk 3]
```diff
@@ -1270,6 +1283,10 @@ ContextImplPtr MemCpyCommandHost::getContext() const {
   return detail::getSyclObjImpl(Queue->get_context());
 }
 
+QueueImplPtr MemCpyCommandHost::getWorkerQueue() const {
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/commands.cpp:1286]: Can reference to QueueImplPtr be returned here to avoid unnecessary copies?

Comment (sergey-semenov) [reply sycl/source/detail/scheduler/commands.cpp:1286]: Sure, this was done this way mostly to mirror the existing `getContext`/`getQueue` methods. Do you mind if all of these are modified in a separate patch?

File: sycl/unittests/scheduler/InOrderQueueDeps.cpp (hunk 2 above)
Comment (sergey-semenov) [reply sycl/unittests/scheduler/InOrderQueueDeps.cpp:128]: Haven't found a good way to verify that with the other command types in unit tests, but I have opened a PR with an E2E test that covers other cases: https://github.com/intel/llvm-test-suite/pull/79

File: sycl/source/detail/scheduler/commands.cpp (hunk 1 above)
Comment (sergey-semenov) [reply sycl/source/detail/scheduler/commands.cpp]: Done

**Review by s-kanaev — APPROVED** [review]

File: sycl/source/detail/scheduler/commands.cpp (hunk 3 above)
Comment (romanovvlad) [reply sycl/source/detail/scheduler/commands.cpp:1285]: I do not mind.

**Review by s-kanaev — APPROVED** [review]

**Pull request merged**